client/security/src/main/java/org/eclipse/kapua/client/security/amqpclient/Client.java

```diff
@@ -73,12 +77,61 @@ public TextMessage createTextMessage() throws JMSException {
         return session.createTextMessage();
     }
 
-    public boolean isConnected() {
+    public void stop() throws JMSException {
+        active = false;
+        disconnect();
+    }
+
+    private void disconnect() throws JMSException {
+        if (connection != null) {
+            connection.close();
+        }
+    }
+
+    private void connect() {
+        int connectAttempt = 0;
+        while (active && !isConnected()) {
+            logger.info("Connect attempt {}...", connectAttempt);
+            try {
+                logger.info("Service client {} - restarting attempt... {}", this, connectAttempt);
+                disconnect();
+                connection = connectionFactory.createConnection();
+                connection.setExceptionListener(exceptionListener);
+                connection.setClientID(clientId);
+                connection.start();
+                session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+                logger.info("AMQP client binding request sender to: {}", requestAddress);
+                logger.info("AMQP client binding message listener to: {}", replyAddress);
+                consumer = session.createConsumer(session.createQueue(replyAddress));
+                consumer.setMessageListener(clientMessageListener);
+                producer = session.createProducer(session.createQueue(requestAddress));
+                clientMessageListener.init(session, producer);
+                connectionStatus = true;
```

Comment:
Is there not a way to read the connection status from one of these objects instead of managing it ourselves?  Should it be set to false on exception?

Reply:
Unfortunately no. The javax.jms.Connection has no connect flag o isConnect method.